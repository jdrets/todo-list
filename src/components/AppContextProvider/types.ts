import React from 'react'

export interface TasksTypes {
  status: string
  priority: string
  dueDate?: string
  title: string
  description: string
}

export interface AppContextProviderTypes {
  tasks: TasksTypes[]
  setTasks: React.SetStateAction<any>
  newTaskModal: boolean
  setNewTaskModal: React.SetStateAction<any>
}
