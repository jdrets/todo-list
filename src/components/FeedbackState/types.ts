import { ReactElement } from 'react'

export interface FeedbackStateTypes {
  illustration: ReactElement
  title: string
  button?: ReactElement
}
