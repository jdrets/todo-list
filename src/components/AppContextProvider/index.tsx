import React, { FunctionComponent, useState, createContext, useEffect } from 'react'
import _ from 'lodash'

import TaskService from '../../services/taskService'
import { AppContextProviderTypes } from './types'

export const PageContext = createContext({} as AppContextProviderTypes)

const AppContextProvider: FunctionComponent<any> = ({ children, value = {} }) => {
  const [tasks, setTasks] = useState([])
  const [showCreateTaskModal, setShowCreateTaskModal] = useState(false)

  useEffect(() => {
    const storage = TaskService.get()
    setTasks(storage)
  }, [])

  const updateContextWithNewTask = (task) => {
    const newTasks = _.clone(tasks)
    newTasks.push(task)
    setTasks(_.reverse(newTasks))
  }

  const contextValues = {
    tasks,
    showCreateTaskModal,
    updateContextWithNewTask,
    setShowCreateTaskModal,
    ...value
  }

  return (
    <PageContext.Provider value={contextValues}>{children}</PageContext.Provider>
  )
}

export default AppContextProvider
