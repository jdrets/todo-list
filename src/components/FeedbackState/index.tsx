import React, { FunctionComponent } from 'react'

import { FeedbackStateTypes } from './types'
import {
  Wrapper,
  IllustrationWrapper,
  TitleWrapper,
  Container
} from './styles'

const FeedbackState: FunctionComponent<FeedbackStateTypes> = ({
  illustration,
  title,
  button
}) => {
  return (
    <Wrapper data-testid="empty-state">
      <Container>
        <IllustrationWrapper>{illustration}</IllustrationWrapper>
        <TitleWrapper>{title}</TitleWrapper>
        {button && button}
      </Container>
    </Wrapper>
  )
}

export default FeedbackState
