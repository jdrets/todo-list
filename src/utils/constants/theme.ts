const palette = {
  purpleDark: '#4B00BC',
  purple: '#6600FF',
  purpleMedium: '#8533FF',
  purpleLight: '#E3D0FF',
  green: '#59BC6F',
  greenLight: '#E6F5E2',
  red: '#EB5757',
  redLight: '#FCE4E4',
  yellow: '#DD9900',
  yellowLight: '#FEF1D8',
  white: '#FFFFFF',
  black: '#000000',
  gray: '#898989',
  grayLight: '#E0E0E0'
}

const theme = {
  colors: {
    primaryColor: palette.purple,
    primaryColorMedium: palette.purpleMedium,
    primaryColorLight: palette.purpleLight,
    primaryColorDark: palette.purpleDark,
    ...palette
  },
  layout: {
    breakpoints: {
      medium: '768px'
    },
    mobile: {
      width: '100%',
      padding: '0 24px'
    },
    desktop: {
      width: '1050px'
    }
  }
}

export default theme
