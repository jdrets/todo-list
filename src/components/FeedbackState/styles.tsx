import styled from 'styled-components'

import theme from '../../utils/constants/theme'

export const Wrapper = styled.section`
  align-items: center;
  display: flex;
  height: 100%;
  justify-content: center;
  left: 0;
  position: absolute;
  top: 0;
  width: 100%;
`

export const Container = styled.div`
  align-items: center;
  display: flex;
  flex-flow: column;
  justify-content: center;
`

export const TitleWrapper = styled.div`
  color: ${theme.colors.gray};
  font-size: 24px;
  font-weight: 500;
  padding: 18px 0 24px;
  text-align: center;
`

export const IllustrationWrapper = styled.div`
  fill: ${theme.colors.gray};
  width: 138px;
`
