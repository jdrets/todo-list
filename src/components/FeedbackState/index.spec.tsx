import React from 'react'
import { ThemeProvider } from 'styled-components'

import { render } from '@testing-library/react'

import FeedbackState from '.'
import theme from '../../utils/constants/theme'
import AppContextProvider from '../../components/AppContextProvider'

const setup = (mockData = {}) => {
  const mock = {
    illustration: <div>dummy illustration</div>,
    title: 'dummy title',
    button: <button>dummy button</button>
  }
  return render(
    <ThemeProvider theme={theme}>
      <AppContextProvider>
        <FeedbackState {...mock} {...mockData} />
      </AppContextProvider>
    </ThemeProvider>
  )
}

describe('<FeedbackState />', () => {
  test('should match with snapshot', () => {
    const { container } = setup()

    expect(container).toMatchSnapshot()
  })

  test('should not render button when button prop is null', () => {
    const mockData = {
      button: null
    }
    setup(mockData)

    const button = document.getElementsByTagName('button')[0]

    expect(button).toBeFalsy()
  })
})
