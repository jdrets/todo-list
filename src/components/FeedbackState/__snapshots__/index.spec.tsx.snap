// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FeedbackState /> should match with snapshot 1`] = `
<div>
  <section
    class="sc-bdnylx dOCMpk"
    data-testid="empty-state"
  >
    <div
      class="sc-gtssRu ljqUJc"
    >
      <div
        class="sc-hKFyIo iMcohj"
      >
        <div>
          dummy illustration
        </div>
      </div>
      <div
        class="sc-dlnjPT jAIsa-d"
      >
        dummy title
      </div>
      <button>
        dummy button
      </button>
    </div>
  </section>
</div>
`;
